### DISCUSSION TITLE
What is Red Knot?

### DISCUSSION THREAD

**Discussion by denialofsandwich** [astral-sh/ruff#14917] [Q&A] [answered]
Hey,

first of all: I really love ruff and you guys did an awesome job here!
Today I stumbled over the mysterious term "red knot", which seems to be a big release or exciting feature that is currently in development, but I wasn't able to exactly find out what it's going to be. I searched in the Discussions, Issues and Documentation, but no full explanation.

Can someone help me out here? :D

**Comment by trag1c** [comment, marked as answer]
The short answer is: Astral's work-in-progress type checker
